README.rst

```diff
@@ -22,6 +22,24 @@ To learn more about SARIF and find resources for working with it, you can visit
 
 The source code is available at https://github.com/microsoft/sarif-python-om.
 
+How to use
+==========
+
+.. code-block:: python
+
+  import cattrs
+  import json
+  import sys
+
+  from sarif_om import SarifLog
+
+
+  with open(sys.argv[1]) as fp:
+      data = json.load(fp)
+
+  sarif_log = cattrs.structure(data, SarifLog)
```

Comment:
Thanks a lot for this pull request and It helps me a lot for better using this library as the changes provide much better type hint and a better way to serialize/deserialize SARIF report.

For the serialization/deserialization, I think just using the default cattrs converter is not enough. Here are the missing points:
1. When serialization to JSON, the property name should be converted to the one in the SARIF JSON schema, e.g. `physical_location` => `physicalLocation`
2. Properties with default values should be ommited.

I finally came up with a solution for that by using a custom cattrs converter. The converter will convert the property name to the JSON schema one and omit the default values:
```python
from attrs import has, fields
from cattrs import Converter
from cattrs.gen import make_dict_unstructure_fn, make_dict_structure_fn, override


def _generate_sarif_report_cattrs_converter() -> Converter:
    converter = Converter()

    def _unstructure(cls):
        return make_dict_unstructure_fn(
            cls,
            converter,
            **{
                a.name: override(
                    rename=a.metadata["schema_property_name"],
                    omit_if_default=True,
                )
                for a in fields(cls)
            },
        )

    def _structure(cls):
        return make_dict_structure_fn(
            cls,
            converter,
            **{
                a.name: override(
                    rename=a.metadata["schema_property_name"],
                )
                for a in fields(cls)
            },
        )

    converter.register_unstructure_hook_factory(has, _unstructure)
    converter.register_structure_hook_factory(has, _structure)
    return converter
```

Here are examples to do de/serialization:

- Serialization:

```python
from sarif_om import *

converter = _generate_sarif_report_cattrs_converter()

sarif_log = SarifLog(
    version="2.1.0",
    runs=[
        Run(
            tool=Tool(
                driver=ToolComponent(
                    name="My Static Analyzer",
                    version="1.0.0",
                    rules=[],
                )
            ),
            results=[
                Result(
                    rule_id="R001",
                    message=Message(text="Use of uninitialized variable 'x'"),
                    locations=[
                        Location(
                            physical_location=PhysicalLocation(
                                artifact_location=ArtifactLocation(
                                    uri=f"file:///path/to/file",
                                )
                            )
                        )
                    ],
                )
            ],
        )
    ],
)

converter.unstructure(sarif_log, SarifLog)
# Then use json.dumps() to get JSON...
```

Will return:

  ```python
  {'runs': [{'tool': {'driver': {'name': 'My Static Analyzer',
       'version': '1.0.0'}},
     'results': [{'message': {'text': "Use of uninitialized variable 'x'"},
       'locations': [{'physicalLocation': {'artifactLocation': {'uri': 'file:///path/to/file'}}}],
       'ruleId': 'R001'}]}],
   'version': '2.1.0'}
  ```

- Deserialization
   
```python
sarif_log_dict = {
    "runs": [
        {
            "tool": {"driver": {"name": "My Static Analyzer", "version": "1.0.0"}},
            "results": [
                {
                    "message": {"text": "Use of uninitialized variable 'x'"},
                    "locations": [
                        {
                            "physicalLocation": {
                                "artifactLocation": {"uri": "file:///path/to/file"}
                            }
                        }
                    ],
                    "ruleId": "R001",
                }
            ],
        }
    ],
    "version": "2.1.0",
}

converter.structure(sarif_log_dict, SarifLog)
```

Will return:

  ```python
  SarifLog(runs=[Run(tool=Tool(driver=ToolComponent(name='My Static Analyzer', associated_component=None, contents=['localizedData', 'nonLocalizedData'], dotted_quad_file_version=None, download_uri=None, full_description=None, full_name=None, global_message_strings=None, guid=None, information_uri=None, is_comprehensive=False, language='en-US', localized_data_semantic_version=None, locations=[], minimum_required_localized_data_semantic_version=None, notifications=[], organization=None, product=None, product_suite=None, properties=None, release_date_utc=None, rules=[], semantic_version=None, short_description=None, supported_taxonomies=[], taxa=[], translation_metadata=None, version='1.0.0'), extensions=[], properties=None), addresses=[], artifacts=[], automation_details=None, baseline_guid=None, column_kind=None, conversion=None, default_encoding=None, default_source_language=None, external_property_file_references=None, graphs=[], invocations=[], language='en-US', logical_locations=[], newline_sequences=['\r\n', '\n'], original_uri_base_ids=None, policies=[], properties=None, redaction_tokens=[], results=[Result(message=Message(arguments=[], id=None, markdown=None, properties=None, text="Use of uninitialized variable 'x'"), analysis_target=None, attachments=[], baseline_state=None, code_flows=[], correlation_guid=None, fingerprints=None, fixes=[], graph_traversals=[], graphs=[], guid=None, hosted_viewer_uri=None, kind='fail', level='warning', locations=[Location(annotations=[], id=-1, logical_locations=[], message=None, physical_location=PhysicalLocation(address=None, artifact_location=ArtifactLocation(description=None, index=-1, properties=None, uri='file:///path/to/file', uri_base_id=None), context_region=None, properties=None, region=None), properties=None, relationships=[])], occurrence_count=None, partial_fingerprints=None, properties=None, provenance=None, rank=-1.0, related_locations=[], rule=None, rule_id='R001', rule_index=-1, stacks=[], suppressions=[], taxa=[], web_request=None, web_response=None, work_item_uris=[])], run_aggregates=[], special_locations=None, taxonomies=[], thread_flow_locations=[], translations=[], version_control_provenance=[], web_requests=[], web_responses=[])], version='2.1.0', schema_uri=None, inline_external_properties=[], properties=None)
  ```

Reply:
@funjim This is great! Can you open a pull request against my branch in my repository?
I'd be happy to integrate it to mine and merge it into this proposal! :bow:

Reply:
> @FunJim This is great! Can you open a pull request against my branch in my repository? I'd be happy to integrate it to mine and merge it into this proposal! 🙇

Sure, please see https://github.com/multani/sarif-python-om/pull/1